Source/UI/NewUI/PaymentForm/View/SecurityCodeView.swift

```diff
@@ -39,20 +45,38 @@ public final class SecurityCodeView: UIView {
     codeInputView.update(style: self.style)
   }
 
-  private func updateErrorViewStyle(isHidden: Bool, textfieldText: String?) {
+  // TODO: integrate with payment vc when card view is finished
+  func updateCardSchema(cardSchema: Card.Scheme) {
+    supportedSchema = cardSchema
+  }
+
+  private func updateErrorViewStyle(isHidden: Bool, textfieldText: String?){
     style?.error?.isHidden = isHidden
     style?.textfield.text = textfieldText ?? ""
     codeInputView.update(style: style)
   }
+
+  func validateSecurityCode(with text: String?) {
+    guard let text = text?.filter { !$0.isWhitespace }, !text.isEmpty else {
+      updateErrorViewStyle(isHidden: false, textfieldText: text)
```

Comment:
Does this mean we don't show error for no input but we do as content is inserted?

Reply:
This is called in `textFieldShouldEndEditing`. if user select textfield then leave it without editing then we show error message.